### PR TITLE
Msm inventory readlist tests

### PR DESCRIPTION
**Pull request by manpreet85**
I have made these changes as per the unknown user test in usertest.go. 

I am getting errors . To see the screen shot of my vscode window, please paste the link in your browser.

https://user-images.githubusercontent.com/15224181/85215717-acafe700-b3d0-11ea-81d5-486b568a054f.png

### CODE REVIEW

**Comment by Jim3Things** [comment]
I sent mail offline about the error.

**Comment by manpreet85** [comment]
Requested changes made and test passed. i.e. negative test of GET to look for rack3 matches the actual response code.

**Comment by manpreet85** [comment]
When I changed it to Rack2 it failed. :)

**Comment by manpreet85** [comment]
Thank you Jim

File: internal/services/frontend/DBInventory.go [hunk 1]
```diff
@@ -68,3 +68,16 @@ func (m *DBInventory) Scan(action func(entry string) error) error {
...
 }
+
+func (m *DBInventory) Get(rackid string) (*pb.ExternalRack, error) {
+
+	m.Mutex.Lock()
+	defer m.Mutex.Unlock()
+
+	r, ok := m.Racks[rackid]
+	if !ok {
+		return nil, NewErrRackNotFound("Rack3")
```
Comment (Jim3Things) [inline internal/services/frontend/DBInventory.go:79]: The argument wants to be the rackid variable, not a literal string.

File: internal/services/frontend/DBInventory_test.go [hunk 2]
```diff
@@ -23,15 +24,56 @@ func TestInventoryListRacks(t *testing.T) {
...
 
-	log.Printf("error: %v", err)
+	assert.Nil(t, err)
+}
+
+func TestInventoryListRead(t *testing.T) {
+	unit_test.SetTesting(t)
+
+	// response := doLogin(t, randomCase(adminAccountName), adminPassword, nil)
+
```
Comment (Jim3Things) [inline internal/services/frontend/DBInventory_test.go:44]: Delete this.  In general, we should not check in commented out code.

File: internal/services/frontend/DBInventory_test.go [hunk 3]
```diff
@@ -23,15 +24,56 @@ func TestInventoryListRacks(t *testing.T) {
...
+	response := doHTTP(request, nil)
+
+	assert.Equal(t, http.StatusOK, response.StatusCode, "Handler returned unexpected error: %v", response.StatusCode)
+
+	rack := &pb.ExternalRack{}
+	err := getJsonBody(response, rack)
+	assert.Nilf(t, err, "Failed to convert body to valid json.  err: %v", err)
+
+	assert.Equal(t, "application/json", strings.ToLower(response.Header.Get("Content-Type")))
+	//assert.Equal(t, "1", response.Header.Get("ETag"))
```
Comment (Jim3Things) [inline internal/services/frontend/DBInventory_test.go:57]: Same here - delete this.

File: internal/services/frontend/DBInventory_test.go [hunk 4]
```diff
@@ -23,15 +24,56 @@ func TestInventoryListRacks(t *testing.T) {
...
+	assert.Equal(t, 2, len(rack.Blades))
+	_, ok := rack.Blades[1]
+	assert.True(t, ok, "Blade 1 not found")
+	_, ok = rack.Blades[2]
+	assert.True(t, ok, "Blade 2 not found")
+
+	// doLogout(t, randomCase(adminAccountName), response.Cookies())
```
Comment (Jim3Things) [inline internal/services/frontend/DBInventory_test.go:64]: And here - delete this.

File: internal/services/frontend/DBInventory_test.go [hunk 5]
```diff
@@ -23,15 +24,56 @@ func TestInventoryListRacks(t *testing.T) {
...
+	unit_test.SetTesting(t)
+
+	request := httptest.NewRequest("GET", fmt.Sprintf("%s%s", baseURI, "/api/racks/Rack3"), nil)
+	request.Header.Set("Content-Type", "application/json")
+
+	response := doHTTP(request, nil)
+	// body, err := getBody(response)
+	// t.Log(string(body))
+
+	// assert.Nil(t, err)
```
Comment (Jim3Things) [inline internal/services/frontend/DBInventory_test.go:76]: And here - delete this.

File: internal/services/frontend/errors.go [hunk 6]
```diff
@@ -4,132 +4,139 @@
...
-        SC: http.StatusForbidden,
-        Base: fmt.Errorf("CloudChamber: user %q is protected and cannot be deleted", name),
-    }
+	return &HTTPError{
+		SC:   http.StatusForbidden,
+		Base: fmt.Errorf("CloudChamber: user %q is protected and cannot be deleted", name),
+	}
+}
+
+func NewErrRackNotFound(name string) *HTTPError {
```
Comment (Jim3Things) [inline internal/services/frontend/errors.go:135]: You should add a comment describing what the intent of the error is (see the other cases above as examples.)

Comment (MikeCJohnsonUk) [reply internal/services/frontend/errors.go:160]: In general, all public names (anything beginning with a capital letter) in the should have a comment of the form

// PublicNameXxx is a xxxx
//
func PublicNameXxx

and if there is no comment, or not one of that form, the lint-er will complain and I'd expect you to see VsCode placing a warning message in the "Problems" section at the bottom looking something like "exported type PublicNameXxxx should have a comment or be unexported". So adding the comment of the expected form will help clean up the issues listed in the Problems panel.

**Pull request merged**